### PR TITLE
Improve code readability for word count.

### PR DESCRIPTION
**Pull request by hiihellox10**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6025)

### CODE REVIEW

File: src/docfx/lib/HtmlUtility.cs [hunk 1]
```diff
@@ -442,11 +430,22 @@ private static int CountWordInText(ReadOnlySpan<char> text)
             return total;
         }
 
+        private static bool IsWordEndChar(char ch)
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs:433]: Does this change caused `IsCJKChar` to be called twice?

Comment (hiihellox10) [reply src/docfx/lib/HtmlUtility.cs:433]: Yes, this is a must repeat. Do you have a better way? Newly added method IsWordEndChar can make the code easier to understand.

Comment (yufeih) [reply src/docfx/lib/HtmlUtility.cs:433]: You can keep the current control flow and only extract methods.

BTW, I was wondering why line 418 and line 427 in the current version behaves differently. If seems to me they are all word separators. Maybe you could help merge them.

Comment (hiihellox10) [reply src/docfx/lib/HtmlUtility.cs:433]: This PR is improve the readability of the code and make the logic of word count statistics easier to understand. The logic has 2 stages: 1, Whether it is the end character of a word(English word). 2, Whether it is a character in a word(English word). The line 418 is stage 1, line 427 is state 2. If I keep the old control flow, it’s readable is not better than now.

Comment (yufeih) [reply src/docfx/lib/HtmlUtility.cs:433]: The current control flow is readable enough to me. For this particular piece of code, I would in favor of avoiding the duplicated check.

Comment (hiihellox10) [reply src/docfx/lib/HtmlUtility.cs:433]: It is enough now. But I like readable code, I have no good way to avoiding the duplicated check now.

**Pull request closed without merging**